### PR TITLE
Update local reference urls to redirect to external if specified.

### PR DESCRIPTION
**Pull request by ryuyu**
@joelverhagen @skofman1 @karann-msft

### CODE REVIEW

**Comment by joelverhagen** [comment]
Can we get unit tests on this?

File: tests/NuGetGallery.Facts/Controllers/PagesControllerFacts.cs [hunk 1]
```diff
@@ -84,5 +85,63 @@ public class TheContactAction : TestContainer
...
+            }
+
+            [Fact]
+            public void ExternalTermsOfUseUrlRedirects()
+            {
+                var customUrl = "http://TermsOfUse";
+                var configuration = GetConfigurationService();
+                
+                var controller = GetController<PagesController>();
+                var result = controller.Terms().Result;
```
Comment (joelverhagen) [inline tests/NuGetGallery.Facts/Controllers/PagesControllerFacts.cs]: use `await` by making the test `async Task`.

File: tests/NuGetGallery.Facts/Controllers/PagesControllerFacts.cs [hunk 2]
```diff
@@ -84,5 +85,63 @@ public class TheContactAction : TestContainer
...
+                Assert.IsType<ViewResult>(result);
+
+                configuration.Current.ExternalTermsOfUseUrl = customUrl;
+                controller = GetController<PagesController>();
+                result = controller.Terms().Result;
+
+                Assert.IsType<RedirectResult>(result);
```
Comment (joelverhagen) [inline tests/NuGetGallery.Facts/Controllers/PagesControllerFacts.cs]: This should be a different "Fact" (i.e. test)

Comment (joelverhagen) [reply tests/NuGetGallery.Facts/Controllers/PagesControllerFacts.cs]: In other words, one act+assert per test

File: src/NuGetGallery/Controllers/PagesController.cs [hunk 3]
```diff
@@ -46,6 +50,11 @@ public virtual ActionResult Page(string pageName)
         [HttpGet]
         public virtual ActionResult About()
         {
+            if (!String.IsNullOrEmpty(_galleryConfigurationService.Current.ExternalAboutUrl))
```
Comment (skofman1) [inline src/NuGetGallery/Controllers/PagesController.cs]: string instead of String everywhere

**Comment by scottbommarito** [comment]
@ryuyu @skofman1 @karann-msft 
What's the status of this?

**Comment by skofman1** [comment]
@ryuyu , is this still relevant? Which issue does this resolve?

**Pull request closed without merging**